### ISSUE TITLE
Set up Travis for CI/CD

### ISSUE THREAD

**Issue by yebrahim** [vocatwk/CodeU-Nemo#47]
In a nutshell:
**Continuous Integration (CI):** The practice of merging your incremental changes into the codebase frequently. This practically implies having your code built and tested automatically when trying to merge your code (creating a pull request). This prevents checking in code that breaks the product.

**Continuous Delivery (CD):** The practice of deploying the latest changes to production automatically upon checking in code.
More: https://stackoverflow.com/a/48477591/1768702.

**Travis:** Popular CI/CD service that is free for open source projects on Github, so great for our use case. More: https://docs.travis-ci.com/. You can basically opt in with your Github account (project admin only can do this), then add a travis.yml file to the repository root with the steps to build and test the code.

Setting up a CI/CD solution is essential for working on quality code. If anyone is interested we can discuss this here.

**Comment by vocatwk** [comment]
Sounds cool. So does it do it like how we do it with the mvn appengine:devserver command automatically whenever we create a request?

**Comment by vocatwk** [comment]
and by pushing to production, do you mean deploying to appengine in our case?

**Comment by yebrahim** [comment]
Well, we wouldn't need to start the dev server since we just want to run the unit tests. There's probably a command to do that in Maven, we'll have to find it out.
Yup, pushing to production would mean deploying to AppEngine in our case.

**Comment by vocatwk** [comment]
Cool. Thanks!

**Comment by lealgamb** [comment]
This seems really helpful. I'm just a little confused about how to set it up?

**Comment by yebrahim** [comment]
First, @vocatwk needs to enable Travis for this repository, since she's admin.

@vocatwk can you take care of that? You'll need to sign into https://travis-ci.com/ with your Github credentials, then choose this repository to enable Travis, and you're done.

@lealgamb after that's done, we'll need to figure out how to write a travis.yml file with the configs needed for setting up the build environment and starting the build and tests. Travis basically loans us a VM with a preset of packages/libraries installed, for example, we can ask for a Java environment VM. [This page](https://docs.travis-ci.com/user/languages/java/) looks like a good start.

**Comment by vocatwk** [comment]
Sure thing. Just enabled it for this repository.

**Comment by yebrahim** [comment]
Great, thanks!
@lealgamb, you can now create a branch, add a travis.yml file (see the page I liked above), and push the branch to origin. New PRs using this branch should now have a Travis status associated with each commit. You can play around with a new PR on this branch, then when it's ready you can ask for the review.

**Comment by lealgamb** [comment]
Awesome, thanks! Once that PR is merged will all branches that are based off the master have the travis.yml file?

**Comment by yebrahim** [comment]
Yes, any branch that has the travis.yml file will cause Travis to kick off a build. If you merge the changes into master, that means any commit going into master (including merge commits caused by PRs) will kick off Travis builds as well.
